app/dns/dns_test.go

```diff
@@ -0,0 +1,61 @@
+package dns
+
+// import (
+// 	"net"
+// 	"testing"
+
+// 	mockdns "github.com/foxcpp/go-mockdns"
+// 	"github.com/stretchr/testify/assert"
+// )
+
+// var mockDNSResolver *mockdns.Server
```

Comment:
This library I use for mock a DNS Nameserver blows up the dependency list. I am not quite sure, is there a way to add this dependency for the tests but avoid in in `go.sum` for reproxy itself?